internal/services/mysql/mysql_flexible_server_resource.go

```diff
@@ -237,6 +237,13 @@ func resourceMysqlFlexibleServer() *pluginsdk.Resource {
 				ValidateFunc: privatezones.ValidatePrivateDnsZoneID,
 			},
 
+			"public_network_access": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				Default:      servers.EnableStatusEnumEnabled,
```

Comment:
Per @neil-yechenwei comment, if Azure doesn't default this consistently to Enabled, we shouldn't default this either as it'll break people's configuration for those who have it setup differently. I believe we can just do Optional/Computed. It shouldn't be a security risk since it normally defaults to True so people removing this from their config won't expect it to turn off

Reply:
Can you also leave an `// NOTE: O + C:` note for why it's Optional/Computed

Reply:
Done!